src/client/automation/playback/click/index.js

```diff
@@ -222,21 +222,39 @@ export default class ClickAutomation {
         return focusAndSetSelection(elementForFocus, simulateFocus, this.caretPos);
     }
 
+    static _getElementForClick (mouseDownElement, topElement, mouseDownElementParentNodes) {
+        var topElementParentNodes = domUtils.getParents(topElement);
+        var areElementsSame       = domUtils.isTheSameNode(topElement, mouseDownElement);
+
+        // NOTE: Mozilla Firefox always skips click, if an element under cursor has been changed after mousedown.
+        if (browserUtils.isFirefox)
+            return areElementsSame ? mouseDownElement : null;
+
+        if (!areElementsSame) {
+            if (mouseDownElement.contains(topElement) && !domUtils.isEditableFormElement(topElement))
+                return mouseDownElement;
+
+            if (topElement.contains(mouseDownElement))
+                return topElement;
+
+            // NOTE: If elements are not in parent relationship, non-ff browsers
+            // are raising`click` event for their common parent.
+            return arrayUtils.getCommonElement(topElementParentNodes, mouseDownElementParentNodes);
+        }
+
+        // NOTE: If target element and top element are same, non-FF browsers are dispatching `click`
+        // event on condition, that target element hasn't changed it position in DOM after mousedown.
```

Comment:
```
        // NOTE: In case the target element and the top element are the same, non-FF browsers are dispatching the 'click'
        // event if the target element hasn't changed its position in the DOM after mousedown.
```